bundles/http/Citrix/DocuSign/73f4e3b3-e2f3-43b3-9ce6-7f12da7b714c.js

```diff
@@ -341,7 +341,7 @@ async function syncTemplates(client, dataStore, account_id, latestSynchronizatio
     } while (start_position < totalSetSize)
 }
 
-async function syncEnvelopes(client, dataStore, account_id, userids, latestSynchronizationTime = null) {
+async function syncEnvelopes(client, dataStore, account_id, userids, dataUpdateAfterAction, latestSynchronizationTime = null) {
```

Comment:
I'm confused about the variable name and how you implemented it.

I suppose it SHOULD work like this: When `dataUpdateAfterAction = false` then the whole `do-while cycle` wouldn't run.
But it run. Because 
```js
do {
  console.log('yes, this is logged')
} while (false)
```

Are you sure that your code works as you expect?

Reply:
Hi @ondrejkralik ,
The variable name `dataUpdateAfterAction` is used to handle the no of page traversed for an api response.

- When the Sync is running(Full Sync or Incremental Sync), then it should traverse all the pages of the api response therefore we are passing dataUpdateAfterAction as false in fullSync and incrementalSync function.
- When the service action is executed, then for dataUpdateAfterAction only first page data of the api response should  be stored in cache therefore we are passing `dataUpdateAfterAction` as true. In this case loop will run only once.